test/functional/feature_bind_extra.py

```diff
@@ -37,37 +37,41 @@ def setup_network(self):
         loopback_ipv4 = addr_to_hex("127.0.0.1")
 
         # Start custom ports by reusing unused p2p ports
-        port = p2p_port(self.num_nodes)
+        def extra_port():
```

Comment:
@w0xlt was also modifying these in a different PR, I also considered a similar solution, but suggested using an iterator and lambdas instead https://github.com/bitcoin/bitcoin/pull/33231#discussion_r2296873010